Content.Client/Alerts/ClientAlertsSystem.cs

```diff
@@ -1,16 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.Alert;
+using JetBrains.Annotations;
 using Robust.Client.GameObjects;
+using Robust.Client.Player;
 using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
 
-namespace Content.Client.Alerts
+namespace Content.Client.Alerts;
+
+[UsedImplicitly]
+internal class ClientAlertsSystem : AlertsSystem
 {
-    internal class ClientAlertsSystem : EntitySystem
+    public AlertOrderPrototype? AlertOrder { get; set; }
+
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public event EventHandler? ClearAlerts;
+    public event EventHandler<IReadOnlyDictionary<AlertKey, AlertState>>? SyncAlerts;
+
+    public override void Initialize()
     {
-        public override void Initialize()
-        {
-            base.Initialize();
+        base.Initialize();
+
+        SubscribeLocalEvent<AlertsComponent, PlayerAttachedEvent>((_, component, _) => PlayerAttached(component));
+        SubscribeLocalEvent<AlertsComponent, PlayerDetachedEvent>((_, _, _) => PlayerDetached());
+
+        SubscribeLocalEvent<AlertsComponent, ComponentHandleState>(ClientAlertsHandleState);
+        SubscribeLocalEvent<AlertsComponent, ComponentShutdown>((_, _, _) => PlayerDetached());
+
+        AlertOrder = _prototypeManager.EnumeratePrototypes<AlertOrderPrototype>().FirstOrDefault();
+        if (AlertOrder == null)
```

Comment:
It seems really weird to define a class of prototypes, solely so that you can randomly load in a single instance and use it. Or are these prototypes also used elsewhere for some other purpose?

If the ordering really needs to be defined in yaml somewhere, I'm not sure  what the best solution is, but even just giving a priority-value to the individual `AlertPrototype`s seems to make more sense?

Reply:
The whole ordering system is super jank, you can see how much crap gets sent through `SyncUpdateControls` in the ui just to order the controls at the last minute. I was hoping not to have to *completely* recode all this, but it is probably better if I do. The sorting should happen much earlier in the update path. Alerts need a unique id, so i can use alphabetical sorting to stabilize the priority sorting during the edge case where multiple `AlertPrototype`s define the same priority.

Reply:
NOPE, not touching this mess.